lib/explorer/series.ex

```diff
@@ -2244,6 +2244,71 @@ defmodule Explorer.Series do
   @spec pow(left :: Series.t() | number(), right :: Series.t() | number()) :: Series.t()
   def pow(left, right), do: basic_numeric_operation(:pow, left, right)
 
+  @euler_number 2.718281828459045
+
+  @doc """
+  Calculates the natural logarithm.
+
+  The resultant series is going to be of dtype `:float`.
+  See `log/2` for passing a custom base. 
+
+  ## Supported dtypes
+
+    * `:integer`
+    * `:float`
+
+  ## Examples
+
+      iex> s = Explorer.Series.from_list([1, 2, 3, nil, 4])
+      iex> Explorer.Series.log(s)
+      #Explorer.Series<
+        Polars[5]
+        float [0.0, 0.6931471805599453, 1.0986122886681098, nil, 1.3862943611198906]
+      >
+
+  """
+  @doc type: :element_wise
+  @spec log(argument :: Series.t()) :: Series.t()
```

Comment:
Instead of having a default value, I think we should really call `log/1`. This will help us avoid precision issues if we support multiple float types in the future and I believe the natural log may be more optimized. For example, it seems Rust implements all other logs in favor of `ln`: https://doc.rust-lang.org/src/std/f64.rs.html#428-430